lib/client.js

```diff
@@ -132,6 +135,14 @@ Client.prototype.getJwtFromUrl = function () {
 };
 
 /**
+ * Attempts to replace the JWT in the url with the new JWT passed in the function.
+ * @param {string} newJwt
+ */
+Client.prototype.replaceJWTInUrl = function (newJwt) {
+  window.location.hash = '/?jwt=' + newJwt;
```

Comment:
This patch will work, I've tested it. You already have most of the code for this. When getting the ID Site view model, stormpath's response view models contains an authorization header that contains a new JWT. You are grabbing it and setting it to the authToken variable in your code. We just need to persist this somehow so on refresh that token gets used.